### PR TITLE
Forms

### PR DESCRIPTION
**Pull request by mikemtzp**
- Create a method called `current_user` in ApplicationController to make current user data available to all controllers.
  - It returns the first user from the database.
- Create forms to perform the following functions:
  - Creates a Post on behalf of the `current_user`.
  - Create a comment on behalf of the `current_user`.
- Allow Users to add likes to Posts.

### CODE REVIEW

File: app/views/posts/show.html.erb [hunk 1]
```diff
@@ -1,10 +1,20 @@
...
-      <p class="section-post__counter">Comments: <%= @post.comments_counter %> | Likes: <%= @post.likes_counter %></p>
-      <p class="section-post__text"><%= @post.text %></p>
+      <div class="title-flex">
+        <button class="back-link"><%= link_to "Comment", new_user_post_comment_path(@user.id, @post.id) %></button>
+        
+        <%= form_with url: user_post_likes_url(@user, @post), method: 'post' do |f| %>
+          <%= f.submit 'Like', class: "back-link" %>
+        <% end %>
+      </div>
+    </div>
```
Comment (codepantha) [inline app/views/posts/show.html.erb:14]: - [x] An error occurs when trying to go to the `/posts` index page and `/posts/:id` show page as shown below 👇 Please fix this bug 🐛 
image

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES ♻️ 

## Hi @mikemtzp 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Good commit history
✔️ No linter errors

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/posts/show.html.erb (hunk 1 above)
Comment (mikemtzp) [reply app/views/posts/show.html.erb:14]: Hi, @codepantha Thanks for your review 😁 

Everything seems to be working on my side with the current code. Any idea of why? 🤔 

image

Comment (codepantha) [reply app/views/posts/show.html.erb:14]: I think in the course of developing the project, you probably have some data you have generated either through rails console or any other method that masked the issue, and as a result, you haven't noticed this problem.

Comment (mikemtzp) [reply app/views/posts/show.html.erb:14]: @codepantha Thanks! I refactored the post controller `show` and `index` methods. The project works while having information in the database.

**Review by mahbubul14 — APPROVED** [review]
# Project Approved :trophy: :tada: 

Hi @mikemtzp , 

Your project is complete!  You have done it exceptionally well. it's time to merge it :100: 
Congratulations! 🎉 

### To Highlight :+1: 

:heavy_check_mark: No linter error.
:heavy_check_mark: Descriptive readme file.
:heavy_check_mark: Well implemented suggested changes.

### Optional suggestions :arrow_down_small: 

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also connect with me on [slack](https://microverse-students.slack.com/team/U01J39GL614)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**